src/main/java/domain/Amount.java

```diff
@@ -0,0 +1,50 @@
+package domain;
+
+import global.exception.ExceptionMessage;
+
+public class Amount {
+    public static final int LOTTO_PRICE = 1000;
+
+    private int amount;
```

Comment:
상태 값이 한번 할당된 뒤부터는 변경되지 않는 것 같은데요. 
그렇다면 `final` 키워드를 사용해서 재할당을 막아보면 어떨까요? 😃

```suggestion
    private final int amount;
```

Reply:
앗 final을 빼먹었군요..ㅠㅠ 

이 경우와는 살짝 다르지만, final과 관련해서 궁금한 점이 하나 있어요!! 보통 메서드 파라미터에서도 값의 변경을 막기 위해 final을 붙이는 것으로 알고 있어요. 

```java
    public Amount(final String inputPrice) {
      ...
    }
```
이론상으로는 값의 변경을 막는다는 이유로 필요하다는 것을 잘 알지만, 실제로는 지금까지 메소드 파라미터에 final을 전부 빼버려서 이슈가 발생한 적은 한번도 없어요. 그래도 "이건 붙이라고 했어!"라는 마인드로 계속 붙이는 것이 좋을까요? 로키의 의견이 궁금합니다!

Reply:
결국 프로그래밍이라는 것은 사람이 하는 행위이기 때문에 항상 실수할 여지가 있다고 생각해요. 
하지만 이런 실수의 요소를 언어단에서 방지해준다면 실수는 애초에 발생하지 않겠죠.  

코드를 나 혼자 관리한다면 나만 잘하면되지만, 함께 일하는 개발자들도 실수하지 않으리란 보장은 없다고 생각해요. 
그렇기 때문에 이런 실수의 여지를 가능하면 코드로 제한해주는 것이 좋다고 생각해요. 

이렇게 언어단에서 변경에 대해서 보장을해준다면 혹시나 실수를하더라고 컴파일 에러로 빠른 시점에 발견하여 대응할 수가 있어 
추후 장애로 이어질 여지가 원천적으로 차단된다고 생각해요. 😃 

무엇이됐던간에 `이건 붙이라고 했어`라는 생각으로 쓰는건 그렇게 바람직하지는 않은 것 같아요. 

> 실제로는 지금까지 메소드 파라미터에 final을 전부 빼버려서 이슈가 발생한 적은 한번도 없어요.

개인적인 생각으로는 유지보수하셨던 시스템의 규모가 그리 크지않거나 혹은 혼자 혹은 소규모의 팀(2~3명 정도) 단위로 유지보수를 했다면 크게 못느껴보셨을 수도 있을 것 같아요.  물론 의식적으로 프로그래밍을 해서 문제가 없었을 수도 있지만 계속 신경을 써야한다는 점이 약간의 피로도로 작용할 수도 있을 것 같네요.

Reply:
> 계속 신경을 써야한다는 점이 약간의 피로도로 작용할 수도 있을 것 같네요.
너무 공감됩니다 😂 사람은 항상 실수할 수 있기 때문에, 최대한 언어쪽에서 보장해주는 안전장치가 필요한다는 말씀이 맞는 것 같아요. 감사합니다 ㅎㅎ